api/users/views.py

```diff
@@ -18,11 +21,16 @@ class UserMixin(object):
 
     def get_user(self, check_permissions=True):
         key = self.kwargs[self.node_lookup_url_kwarg]
+        current_user = self.request.user
 
         if key == 'me':
-            return self.request.user
+            if isinstance(current_user, AnonymousUser):
+                raise NotFound
```

Comment:
404s are used for invalid URIs; in this case, the URI is valid, but the response is different depending on whether authentication is provided or not.